### PR TITLE
Feat/task 2

### PR DESCRIPTION
**Pull request by AlexGusew**

### CODE REVIEW

File: src/lesson-2/routers/user/user.router.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import express, { Request } from 'express';
+import validationMiddleware from '../../middlewares/validation';
+import UserDto from './user.dto';
+import * as UserService from './user.service';
+import { User, UserSearchOptions } from './user.types';
+
+const userRouter = express.Router();
+
+userRouter.get('/:id', (req, res) => {
```
Comment (meteorra) [inline src/lesson-2/routers/user/user.router.ts:9]: Usually controllers are split into separate functions

File: src/lesson-2/routers/user/user.service.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+export const getUserBuId = (id: User['id']) => UserDb.getById(id);
+
+export const createNewUser = (newUser: Omit<User, 'id'>) =>
+  UserDb.createUser(newUser);
+
+export const updateUser = (user: PartialUserWithId) => UserDb.updateUser(user);
+
+export const removeUser = (id: User['id']) => UserDb.removeUser(id);
+
+export const getUsers = UserDb.getUsers.bind(UserDb);
```
Comment (meteorra) [inline src/lesson-2/routers/user/user.service.ts:13]: Just from readability perspective, it's better to have all functions declared consistently. Either as an arrow function or using currying

File: src/lesson-2/routers/user/user.router.ts (hunk 1 above)
Comment (AlexGusew) [reply src/lesson-2/routers/user/user.router.ts:9]: Ok, let me update it

File: src/lesson-2/routers/user/user.service.ts (hunk 2 above)
Comment (AlexGusew) [reply src/lesson-2/routers/user/user.service.ts:13]: Will update

**Pull request merged**